src/qibolab/instruments/zhinst/executor.py

```diff
@@ -0,0 +1,781 @@
+            for i, pulse in enumerate(self.sequence[ch]):
+                measurement_groups[i].append((ch, pulse))
+
+        measurement_starts = {}
+        for i, group in measurement_groups.items():
+            starts = np.array([meas.pulse.start for _, meas in group])
+            measurement_starts[i] = max(starts)
+
+        # split all non-measurement channels according to the locations of the measurements
+        sub_sequences = defaultdict(lambda: defaultdict(list))
+        for ch in other_channels:
+            measurement_index = 0
+            for pulse in self.sequence[ch]:
+                if pulse.pulse.finish > measurement_starts[measurement_index]:
+                    measurement_index += 1
+                sub_sequences[measurement_index][ch].append(pulse)
+        if len(sub_sequences) > len(measurement_groups):
+            log.warning("There are control pulses after the last measurement start.")
+
+        return [
+            SubSequence(measurement_groups[i], sub_sequences[i])
+            for i in range(len(measurement_groups))
+        ]
+
+    def experiment_flow(
+        self,
+        qubits: dict[str, Qubit],
+        couplers: dict[str, Coupler],
+        sequence: PulseSequence,
+        options: ExecutionParameters,
+    ):
+        """Create the experiment object for the devices, following the steps
+        separated one on each method:
+
+        Translation, Calibration, Experiment Definition.
+
+        Args:
+            qubits (dict[str, Qubit]): qubits for the platform.
+            couplers (dict[str, Coupler]): couplers for the platform.
+            sequence (PulseSequence): sequence of pulses to be played in the experiment.
+        """
+        self.sequence = self.sequence_zh(sequence, qubits)
+        self.sub_sequences = self.create_sub_sequences(list(qubits.values()))
+        self.calibration_step(qubits, couplers, options)
+        self.create_exp(qubits, options)
+
+    # pylint: disable=W0221
+    def play(self, qubits, couplers, sequence, options):
+        """Play pulse sequence."""
+        self.signal_map = {}
+
+        self.frequency_from_pulses(qubits, sequence)
+
+        self.experiment_flow(qubits, couplers, sequence, options)
+
+        self.run_exp()
+
+        # Get the results back
+        results = {}
+        for qubit in qubits.values():
+            q = qubit.name  # pylint: disable=C0103
+            for i, ropulse in enumerate(self.sequence[measure_channel_name(qubit)]):
+                data = np.array(self.results.get_data(f"sequence{q}_{i}"))
+                if options.acquisition_type is AcquisitionType.DISCRIMINATION:
+                    data = (
+                        np.ones(data.shape) - data.real
+                    )  # Probability inversion patch
+                serial = ropulse.pulse.serial
+                qubit = ropulse.pulse.qubit
+                results[serial] = results[qubit] = options.results_type(data)
+
+        return results
```

Comment:
`sweep()` with zero sweepers should naturally implement `play`, as it is for QM
https://github.com/qiboteam/qibolab/blob/595351782d55a20fa2b8c35104c000092e42f585/src/qibolab/instruments/qm/controller.py#L316-L317

Could you apply the same here?